libraries/rush-lib/src/api/RushCustomTipsConfiguration.ts

```diff
@@ -0,0 +1,94 @@
+import { FileSystem, ITerminal, JsonFile, JsonSchema } from '@rushstack/node-core-library';
+
+import schemaJson from '../schemas/rush-custom-tips.schema.json';
+
+/**
+ * @beta
+ */
+export interface IRushCustomTipsJson {
+  prefix?: string;
+  customTips?: IRushCustomTipItemJson[];
+}
+/**
+ * @beta
+ */
+export interface IRushCustomTipItemJson {
+  id: RushCustomTipId;
+  tip: string;
+  prefix?: string;
+  severity?: RushCustomTipSeverity;
+}
+
+/**
+ * @beta
+ */
+export enum RushCustomTipSeverity {
+  log = 0,
+  warning = 1,
+  error = 2
+}
+
+/**
+ * @beta
+ * TODO: consider making this work with the plugin (i.e., the plugins are able to define their own customizable tips)
+ */
+export type RushCustomTipId = 'PNPM_MISMATCH_DEPENDENCY' | 'ANOTHER_ID_THAT_IS_EXPOSED_TO_CUSTOM';
+
+/**
+ * @beta
+ */
+export class RushCustomTipsConfiguration {
+  private static _jsonSchema: JsonSchema = JsonSchema.fromLoadedObject(schemaJson);
+
+  private readonly _tipMap: Map<RushCustomTipId, IRushCustomTipItemJson>;
+  private readonly _jsonFileName: string;
+
+  public readonly configuration: Readonly<IRushCustomTipsJson>;
+
+  public constructor(configFilename: string) {
+    this._jsonFileName = configFilename;
+    this._tipMap = new Map();
+
+    if (!FileSystem.exists(this._jsonFileName)) {
+      this.configuration = {};
+    } else {
+      this.configuration = JsonFile.loadAndValidate(
+        this._jsonFileName,
+        RushCustomTipsConfiguration._jsonSchema
+      );
+
+      const customTips: IRushCustomTipItemJson[] | undefined = this.configuration?.customTips;
+      if (customTips) {
+        for (const tipItem of customTips) {
+          this._tipMap.set(tipItem.id, tipItem);
```

Comment:
Since the current JSON schema is an array, it allows duplicate entries for the same id. You'll want to add a step here that logs warnings/errors if the same id is listed multiple times.

Reply:
Fixed